content/en/docs/contributing/coding-conventions.md

```diff
@@ -0,0 +1,815 @@
+```
+
+```c
+while (condition)
+        single_statement();
+```
+
+But use braces for the `else block` if the `if block` requires them:
+
+```c
+if (condition) {
+        single_statement();
+        single_statement();
+} else {
+        single_statement();
+}
+```
+
+### Spaces
+
+Generally, put a space after keywords like `if`, `switch`, `case`, `for`, `do`, `while`, `return`.
+
+```c
+if (condition) ✅
+```
+
+```c
+if(condition)  ❌
+```
+
+Also put a space after a comma:
+
+```c
+void func(int arg0, int arg2) ✅
+```
+
+```c
+void func(int arg0,int arg2)  ❌
+```
+
+And around most operators (except unary operators):
+
+```c
+a = b + c;
+```
+
+```c
+if (a == b)
+```
+
+```c
+a = (condition) ? 1 : 2;
+```
+
+For unary operators and structure member operators **do not use** a space:
+
+```c
+p = &var;
+```
+
+```c
+if (!condition)
+```
+
+```c
+a = ~c & b;
+```
+
+```c
+a->field = 0xdeadb0b0;
+```
+
+Do not put spaces around keywords like `sizeof`, `typeof`, `alignof`,`__attribute__` and around (inside) parenthesized expressions:
+
+```c
+s = sizeof(p);    ✅
+```
+
+```c
+s = sizeof ( p ); ❌
+```
+
+```c
+if ( condition )  ❌
+```
+
+Always put the asterisk (\*) for pointer definitions adjacent to the data or function name.
+This is to avoid the misconception that char\* a, b defines two pointers, whereas in fact only a is a pointer.
+
+```c
+char *myfunc(void *p)
+{
+    char *a, *b, c;
+    /* ... */
+
+    return (char *) p;
+}
+```
+
+### Comments
+
+* Always use multi-line comments even for single line comments:
+
+```c
+/* Single line comment */
+if (condition)           ✅
+```
+
+```c
+int var;  /* comment */  ✅
+```
+
+```c
+int var;  // comment     ❌
+```
+
+* Use a separate line to close the comment for actual multi-line comments.
+There is a leading space to align the lines:
+
+```c
+/* This is a multi line comment because it is so incredibly
+ * useful that the contents does not fit into one line
+ */
+```
+
+* Trailing dots are not necessary, unless there is a sentence that you would end with a do.
+Just keep it consistent.
+
+* Use spaces instead of tabs to indent within comments.
+
+* Start comments with capital letters in case of sentences.
+
+### Function Documentation
+
+Functions that are exported by a library or define an otherwise publicly visible interface need to be properly documented in the header file.
+Use [doxygen-style](https://www.doxygen.nl/) comments for this.
+Note the double-asterisk (\*\*) at the beginning:
+
+```c
+/**
+ * Tries to allocate the requested number of bytes from the default memory
+ * allocator. The actual amount of allocated memory can be smaller.
+ *
+ * @param size
+ *   Number of bytes to allocate
+ * @param[out] p
+ *   Pointer to the newly allocated memory on success, NULL otherwise
+ *
+ * @return
+ *   The number of bytes actually allocated
+ */
+size_t try_alloc(size_t size, void **p);
+```
+
+* Use dots when ending a sentence or when a new sentence starts afterwards.
```

Comment:
The only reference I could quickly find on using periods is this: https://softwareengineering.stackexchange.com/a/17769/389332

We can make it so that when using a single sentence, the period is suppressed. But for multiple sentence, I see no point in suppressing the period for the last sentence; it's inconsistent and visually weird.